src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -952,6 +956,25 @@
                                             [server-id (initialize-context)]))
                  :default-server (get-default-server-from-config config)))
 
+(schema/defn ^:always-validate reload-crl-on-change!
+  "Reload the CRL file used by the supplied ssl-context-factory whenever any
+  changes to the file occur."
+  [ssl-context-factory :- InternalSslContextFactory
+   watcher :- (schema/protocol watch-protocol/Watcher)]
+  (when-let [crl-path (.getCrlPath ssl-context-factory)]
+    (let [normalized-crl-path (.getCanonicalPath (fs/file crl-path))]
+      (watch-protocol/add-watch-dir! watcher (fs/parent crl-path)
```

Comment:
For code reuse, I think it would be reasonable for tk-jetty9 to use the tk-fs-watcher and just enhance tk-fs-watcher with an option for not doing the `recursive` directory watching.  The restriction around being unable to set a watcher at a file rather than a directory level, though, seems like a fundamental issue in the Java watcher API implementation.  We might be able to work around that by layering an API on top of the Java FFI / inotify APIs directly but I think that could be a lot of work.  If we decided it were worthwhile to do that rather than waiting for the Java watcher API to hopefully support it directly, might be better to layer that through the tk-filesystem-watcher API as well for better reusability.